CMakeLists.txt

```diff
@@ -0,0 +1,4 @@
+cmake_minimum_required(VERSION 3.9)
+project(NamedType)
```

Comment:
Maybe change this to `project(NamedType CXX)` to indicate that it's a C++ project.